dev/com.ibm.ws.ras.instrument/src/com/ibm/ws/ras/instrument/internal/main/LibertyRuntimeTransformer.java

```diff
@@ -368,6 +393,67 @@ public static byte[] transform(byte[] bytes, boolean skipIfNotPreprocessed) thro
         return result;
     }
 
+    public static byte[] transform(String className, byte[] classBytes, boolean skipIfNotPreprocessed) throws IOException {
+        String methodName = "transform";
+
+        boolean isLoggable = isLoggable(className);
+        boolean isDumpable = tc.isDumpEnabled();
+
+        // The reader is created early and is provided to the class writer
+        // as a write optimization which helps when the writer is optimized
+        // for 'mostly add' transformations.  See the ASM ClassWriter JavaDoc.
+
+        ClassReader classReader = new ClassReader(classBytes);       
+        ClassWriter classWriter = new ClassWriter(classReader, ClassWriter.COMPUTE_MAXS);
+
+        // A class writer is at the bottom of the stack, as it receives all
+        // visit perturbations from the other visitors.  The writer must be
+        // remembered, as it provides the final, transformed class bytes.
+
+        ClassVisitor classVisitor = classWriter;
+
+        // The check and trace visitors are next to last: They validate and display
+        // the perturbed class just before it reaches the writer.
+
+        if ( isDumpable || isLoggable ) {
+            classVisitor = new CheckClassAdapter(classVisitor, false);
+        }
+        if ( isLoggable ) {
+            PrintWriter traceWriter = fileWriter();
+            if ( traceWriter != null ) {
+                classVisitor = new TraceClassVisitor(classVisitor, traceWriter);
+            }
+        }
+        StringWriter sw;
+        if ( isDumpable ) {
+            sw = new StringWriter();
+            classVisitor = new TraceClassVisitor(classVisitor, new PrintWriter(sw));            
+        } else {
+            sw = null;
+        }
+
+        // Just one trace injection class adapter is used.
+        // Contrast this with static trace instrumentation, which
+        // may have a distinct adapter for FFDC injection.
+        
+        LibertyTracingClassAdapter tracingClassAdapter = new LibertyTracingClassAdapter(classVisitor, skipIfNotPreprocessed);
+        try {
+            classReader.accept(tracingClassAdapter, skipDebugData ? ClassReader.SKIP_DEBUG : 0);
```

Comment:
No longer is the code handling the case with the compute max and the throw computer frames.  Is that intentional?  Comment from the previous code:

```
		// Occasionally, frames must be computed.  This is not known until
		// the class is visited, as the circumstances depend on encountering
		// variable manipulating byte-codes before invoking the superclass
		// initializer.
		//
		// The (hopefully rare) occasions are communicated by throwing
    	// 'ComputeRequiredException'.
                
        // First try: Use COMPUTE_MAXS and THROW_COMPUTE_FRAMES in case
        // the flag must be reset.
```